Update upload_liz_genomes.py

Mxrcon opened the pull request:
Hello, I have some suggestions to improve the security and the user experience with this script.

1. `os.listdir()` could make easier to capture the genomes inputs instead of declaring every genome name on script.
2. Declaring username and password at code is considered unsafe for _Good Programming Practices_, maybe with `sys.argv` this security issue could be avoided.  
3. The implementation of some argument parsing method would make the user experience easier and avoid script modifications on every use.

Thanks for your attention, Hope this Issue helps. I'm available to proceed with this modifications if you agreed.

Mxrcon committed:
Update upload_liz_genomes.py

Why declare every genome?  os.listdir() makes it faster.
Declaring username and password at code is unsafe, maybe using sys.argv this issue could be avoided.  

Now takes 3 arguments after python upload_liz_genomes.py Genomes_Path Username Password

Mxrcon committed:
Update upload_liz_genomes.py

Fixing syntax error

abhi18av commented on upload_liz_genomes.py:
```diff
@@ -1,112 +1,18 @@
-"MTB37.2.fastq.gz",
-"MTB45.2.fastq.gz"
-    ]
+genome_list = [ i for i in os.listdir(sys.argv[1]) if "fastq.gz" in i]
```
@Mxrcon , perhaps we could also make this `fastq.gz` pattern as a parameter since there are alternative namings such as `fq.gz` etc.

Mxrcon replied:
OK! adding it to argparser

abhi18av commented on upload_liz_genomes.py:
```diff
@@ -1,112 +1,18 @@
 """
 2. Update the `username` and `password` 
 """
+username = sys.argv[2]
+password = sys.argv[3]
```
I think that `named parameters` are always more helpful than positional parameters. Let's switch to `argparse`.

You can use this PR diff, as a reference on how can you proceed with this.

Mxrcon replied:
OK! I'll code that

abhi18av reviewed (changes requested):
Oi @Mxrcon , obrigado pela atenção!

Eu solicitei algumas mudanças, se você pode implementar, será ótimo.

Mxrcon committed:
Update upload_liz_genomes.py

Added argparser, should we add email as arg? or keep it on script? we're almost doing a complete CLI program

Mxrcon committed:
Update README.md

added arguments

Mxrcon committed:
Update README.md

Mxrcon committed:
Update README.md

Mxrcon committed:
Update README.md

Mxrcon committed:
Update README.md

abhi18av commented on upload_liz_genomes.py:
```diff
@@ -1,148 +1,70 @@
+parser.add_argument("-e","-extension",action="store", help= "File extension from files",required = False, default="fastq.gz")
+parser.add_argument("-u","-username",action='store', help="NCBI - Username",required = True)
+parser.add_argument("-p","-password",action='store', help= "NCBI - Password",required = True)
+
```
```suggestion
parser.add_argument("-d","-ftpdir",action='store', help= "The directory on the NCBI server",required = True)
```

abhi18av commented on upload_liz_genomes.py:
```diff
@@ -1,148 +1,70 @@
+4. Create a directory using FTP, once you've logged in and you have your own address like `emilyncosta_gmail.com_xxxxx` and create a directory where you'll upload submission `africanum`
 """
 
 ftp.cwd("uploads/lizaspinasse_gmail.com_ojGYyyaI")
```
I think we can also take this folder address via args as well. Would you mind adding the `-d` option, I [suggested](https://docs.github.com/en/github/collaborating-with-issues-and-pull-requests/incorporating-feedback-in-your-pull-request) to avoid hardcoding stuff.

Mxrcon replied:
Thank you, I'll implement that and make the last commit

abhi18av reviewed (commented):
Thanks for taking the time to implement those changes @Mxrcon !

We're almost there, I just added a suggestion and after that we're good to go ahead with the merge.

Mxrcon committed:
Update upload_liz_genomes.py

Co-authored-by: Abhinav Sharma <abhi18av@users.noreply.github.com>

Mxrcon committed:
Update upload_liz_genomes.py

Updated ftp folder name on server, it'll create a folder on server with the given name and inside this folder, it'll create another folder with the same name as the input folder.

Mxrcon committed:
Update README.md

abhi18av reviewed (approved):
Muito obrigado Dávi! 🎉

Mxrcon commented:
:smile: De nada!

abhi18av commented:
@Mxrcon now that you've started to work on PRs and in future would contribute to other codebases individually, please remember to be polite, with decency in the way you express the problem/solution.

I highly recommend that you read this
https://www.contributor-covenant.org/version/1/4/code-of-conduct/ which is a standard for anyone wishing to contribute and interact with NF community. https://github.com/nextflow-io/nextflow/blob/22ae730e72925322a0a83fc4c3375c05da53b5b2/CODE-OF-CONDUCT.md 


I request you to please re-phrase the content of this comment  https://github.com/emilyncosta/upload_genomes_via_ftp/pull/1#issue-562155936 and post the updated content here.

Mxrcon commented:
@abhi18av Ok! changed it trying to be more polite and help with the issue and propose changes that would solve this issues, I'll be more polite on the next issues. Thanks for pointing that, also, I'm sorry if I've been rude on my comments.

abhi18av commented:
Tá amigo, obrigado você 😊
